File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        curr = root

        while curr != None:
            if p.val < curr.val:
                if q.val < curr.val:
                    curr = curr.left
                else: 
                    return curr
            elif p.val > curr.val:
                if q.val > curr.val:
                    curr = curr.right
                else:
                    return curr
            else:
                return curr

        return curr
            
            

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int left = 0;
        int right = 0;
        while (right < prices.length) {
            if (prices[left] < prices[right]) {
                max = Math.max(prices[right] - prices[left], max);
                right++;
            } else {
                left = right;
                right++;
            }
        }
        return max;
    }
}
